integrationTests/node/heartbeat_test.go

```diff
@@ -0,0 +1,173 @@
+package node
+
+import (
+	"context"
+	"encoding/hex"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/ElrondNetwork/elrond-go/crypto"
+	"github.com/ElrondNetwork/elrond-go/crypto/signing"
+	"github.com/ElrondNetwork/elrond-go/crypto/signing/kyber"
+	"github.com/ElrondNetwork/elrond-go/crypto/signing/kyber/singlesig"
+	"github.com/ElrondNetwork/elrond-go/integrationTests"
+	"github.com/ElrondNetwork/elrond-go/node/heartbeat"
+	"github.com/ElrondNetwork/elrond-go/p2p"
+	"github.com/ElrondNetwork/elrond-go/sharding"
+	"github.com/stretchr/testify/assert"
+)
+
+var stepDelay = time.Second
+
+// TestHeartbeatMonitorWillUpdateAnInactivePeer test what happen if a peer out of 2 stops being responsive on heartbeat status
+// The active monitor should change it's active flag to false when a new heartbeat message has arrived.
+func TestHeartbeatMonitorWillUpdateAnInactivePeer(t *testing.T) {
+	if testing.Short() {
+		t.Skip("this is not a short test")
+	}
+
+	advertiser := integrationTests.CreateMessengerWithKadDht(context.Background(), "")
+	_ = advertiser.Bootstrap()
+	advertiserAddr := integrationTests.GetConnectableAddress(advertiser)
+	maxUnresposiveTime := time.Second * 2
+
+	monitor := createMonitor(maxUnresposiveTime)
+	nodes, senders, pks := prepareNodes(advertiserAddr, monitor)
+
+	defer func() {
+		_ = advertiser.Close()
+		for _, n := range nodes {
+			_ = n.Close()
+		}
+	}()
+
+	fmt.Println("Delaying for node bootstrap and topic announcement...")
+	time.Sleep(time.Second * 5)
+
+	fmt.Println("Sending first messages from both public keys...")
+	_ = senders[0].SendHeartbeat()
+	_ = senders[1].SendHeartbeat()
+
+	time.Sleep(stepDelay)
+
+	fmt.Println("Checking both public keys are active...")
+	checkReceivedMessages(t, monitor, pks, []int{0, 1})
+
+	fmt.Println("Waiting for max unresponsive time...")
+	time.Sleep(maxUnresposiveTime)
+
+	fmt.Println("Only first pk will send another message...")
+	_ = senders[0].SendHeartbeat()
+
+	time.Sleep(stepDelay)
+
+	fmt.Println("Checking only first pk is active...")
+	checkReceivedMessages(t, monitor, pks, []int{0})
+}
+
+func prepareNodes(
+	advertiserAddr string,
+	monitor *heartbeat.Monitor,
+) ([]p2p.Messenger, []*heartbeat.Sender, []crypto.PublicKey) {
+
+	interactingNodes := 3
+	nodes := make([]p2p.Messenger, interactingNodes)
+	topicHeartbeat := "topic"
+	senders := make([]*heartbeat.Sender, 0)
+	pks := make([]crypto.PublicKey, 0)
+
+	for i := 0; i < interactingNodes; i++ {
+		nodes[i] = integrationTests.CreateMessengerWithKadDht(context.Background(), advertiserAddr)
+		_ = nodes[i].CreateTopic(topicHeartbeat, true)
+
+		isSender := i < 2
```

Comment:
it would be better to have a senderListID, like idxProposers in the other tests.

Reply:
done